pkg/worker/model_status_update_batch_job.go

```diff
@@ -0,0 +1,83 @@
+package worker
+
+import (
+	"context"
+	"fmt"
+
+	faktory "github.com/contribsys/faktory/client"
+	faktory_worker "github.com/contribsys/faktory_worker_go"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/mint-app/pkg/models"
+)
+
+var modelStatusUpdateJobMaxRetry = 2
+
+// ModelStatusUpdateBatchJob is the job the cron job initiates to check if models need a status update.
+// It will batch all child jobs, and when complete it will fire a callback
+func (w *Worker) ModelStatusUpdateBatchJob(ctx context.Context, args ...interface{}) error {
+	helper := faktory_worker.HelperFor(ctx)
+	sugaredLogger := w.Logger.With(zap.Any("JID", helper.Jid()), zap.Any("BID", helper.Bid()), zap.Any(appSectionKey, faktoryLoggingSection))
+	sugaredLogger.Info("Getting collection of model plans that require status checking")
+
+	// TODO: Implement the logic to return the models to check? Or do we check every model plan?
+	modelsToUpdate, err := w.Store.ModelPlanCollection(sugaredLogger, false)
+	if err != nil {
+		sugaredLogger.Error("unable to get model plan collection for the model status update batch job", zap.Error(err))
+		return err
+	}
+	return helper.With(func(cl *faktory.Client) error {
+		return CreateModelStatusUpdateBatch(sugaredLogger, w, cl, modelsToUpdate)
+	})
+}
+
+func CreateModelStatusUpdateBatch(logger *zap.Logger, w *Worker, cl *faktory.Client, modelPlans []*models.ModelPlan) error {
+	batch := faktory.NewBatch(cl)
+	batch.Description = "Check if model status should be updated"
+	batch.Success = faktory.NewJob(modelStatusUpdateBatchJobSuccessName)
+	batch.Success.Queue = criticalQueue
+	// TODO: verify this overides the parent BID
+	sugaredLogger := logger.With(zap.Any("BID", batch.Bid))
+
+	err := batch.Jobs(func() error {
+		for _, plan := range modelPlans {
+			err := CreateModelStatusJobInBatch(sugaredLogger, w, batch, plan)
+			if err != nil {
+				err = fmt.Errorf(" error creating job for ModelStatus Update. modelPlanID %v. Err %w", plan.ID, err)
+				sugaredLogger.Error("issue creating ModelStatusUpdateJob", zap.Error(err))
+				return err
+			}
+		}
+		return nil
+
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+
+}
+
+func CreateModelStatusJobInBatch(logger *zap.Logger, w *Worker, batch *faktory.Batch, plan *models.ModelPlan) error {
+	sugaredLogger := logger.With(zap.Any("modelPlanID", plan.ID))
+	sugaredLogger.Info("creating job for model status update.")
+	job := faktory.NewJob(modelStatusUpdateJobName, plan.ID)
+	//TODO: verify, do we want to just use the critical queue? Or should we generate another queue for this? Or perhaps choose a different existing queue?
```

Comment:
I'm not sure about this either. @ClayBenson94 what do you think?

Reply:
Do you have a list of default queues for this? I'd say I'd be tempted to reserve the critical queue pretty much for Change History (since it needs to run every minute)

If there's a queue just lower than critical (high?) then I think that'd suffice for this kind of stuff, until we figure out any other categories/priorities that make sense!

Reply:
It'd help me if we had a clear definition of what the priorities are intended to handle. I generally agree this feels like a standard task, not a critical one.

Reply:
So! Currently using the critical queue 1. Cron Jobs (Analyze Audit Batch, Translate Audit Batch, This method)  2. Analyzed Audit Jobs all trigger on the critical queue.

The `auditTranslateQueue` is configured for the individual jobs in a Translate Audit Batch, (so it doesn't grab priority from cron jobs).

I do agree we probably need some consensus around queue priorities. This is the current queue priorities
```
	// pull jobs from these queues, in this order of precedence
	mgr.ProcessStrictPriorityQueues(criticalQueue, defaultQueue, auditTranslateQueue, emailQueue)
```

I'd suggest
1. critical queue --> Jobs triggered by cron Jobs (batch jobs?)
2. default --> ? (maybe standard jobs?)
3. auditTranslateQueue --> just audit translations
4. emailQueue --> just email jobs?

perhaps we can do a quick staybehind after standup to get consensus on this?

Reply:
Thank you both, as we discussed, we will leave this for now. I made a ticket to audit queues for translations (MINT-3036)